src/components/DonateButton.jsx

```diff
@@ -149,107 +223,151 @@ class DonateButton extends React.Component {
       modalVisible: true,
       amount: '',
       formIsValid: false,
-    });
+    }, () =>
+      this.pollWallet()
+    );
   }
 
   submit(model) {
     this.donateWithBridge(model);
     this.setState({ isSaving: true });
   }
 
-  async donateWithBridge(model) {
+  donateWithBridge(model) {
     const { currentUser } = this.props;
     const { adminId } = this.props.model;
-    const { account, givethBridge, etherscanUrl, showCustomAddress } = this.state;
+    const { account, givethBridge, etherscanUrl, showCustomAddress, selectedToken } = this.state;
 
     const value = utils.toWei(model.amount);
-
-    const opts = { value, gas: DONATION_GAS, from: account };
-    let method;
-    let donationUser;
-
-    if (showCustomAddress) {
-      // Donating on behalf of another user or address
-      try {
-        const user = await feathersClient.service('users').get(model.customAddress);
-        if (user && user.giverId > 0) {
-          method = givethBridge.donate(user.giverId, adminId, opts);
-          donationUser = user;
-        } else {
-          givethBridge.donateAndCreateGiver(model.customAddress, adminId, opts);
+    const isDonationInToken = selectedToken.symbol !== 'ETH'
+    const tokenAddress = isDonationInToken ? selectedToken.address : 0;
+
+    const _makeDonationTx = async () => {
+      let method;
+      let donationUser;
+      let opts;
+
+      if(isDonationInToken)
+        opts = { from: account, gas: 1000000 };
```

Comment:
this gas is ridiculously high, maybe we can just remove it and let the gas estimator figure it out.

Reply:
Yeah we tried that. It does not work.
It put it on 100.000, it says it uses 84766 but it runs out of gas with the exact number